streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java

```diff
@@ -311,9 +325,98 @@ public void shouldProcessDataFromStoresWithLoggingDisabled() throws InterruptedE
         latch.await(30, TimeUnit.SECONDS);
 
         assertTrue(processorLatch.await(30, TimeUnit.SECONDS));
+    }
+
+    @Test
+    public void shouldRecycleStateFromStandbyTaskPromotedToActiveTaskAndNotRestore() throws Exception {
```

Comment:
Alright I tried to shore up the test to verify that we actually do not restore anything in addition to not closing the store itself. This should be closer to testing the specific behavior pointed out in the ticket